integration/spark/src/test/scala/org/apache/carbondata/spark/testsuite/longstring/VarcharDataTypesBasicTestCase.scala

```diff
@@ -110,6 +114,51 @@ class VarcharDataTypesBasicTestCase extends QueryTest with BeforeAndAfterEach wi
     assert(exceptionCaught.getMessage.contains("its data type is not string"))
   }
 
+  test("cannot alter sort_columns dataType to long_string_columns") {
+    val exceptionCaught = intercept[RuntimeException] {
+      sql(
+        s"""
+           | CREATE TABLE if not exists $longStringTable(
+           | id INT, NAME STRING, description STRING, address STRING, note STRING
+           | ) STORED AS carbondata
+           | TBLPROPERTIES('SORT_COLUMNS'='name, address')
+           |""".
+          stripMargin)
+      sql("ALTER TABLE long_string_table SET TBLPROPERTIES('long_String_columns'='NAME')")
+    }
+    assert(exceptionCaught.getMessage.contains("LONG_STRING_COLUMNS cannot be present in sort columns: name"))
+  }
+
+  test("check compaction after altering range column dataType to longStringColumn") {
+    CarbonProperties.getInstance()
```

Comment:
this property value should be reset after this test case else it will impact later test cases. Take the exiting value, set new value, reset to old value

Reply:
Done